### PR TITLE
Limits buffer memory between GC bridge and write flow.

### PR DESCRIPTION
**Pull request by balazsracz**
Adds LimitedPool, an implementation of a BufferPool that allows only a fixed number of buffers to be allocated.

After this is exhausted, further allocations get blocked until some earlier buffer gets freed. This is similar to FixedPool except the memory is allocated and returned to mainBufferPool, so the overhead on a per-instance basis is much lower.

Adds LimitedPool to the BufferPort component which is responsible for sending gridconnect packets to the Hub's write flow. A link-time configuration option defines whether we should limit the number of filled-in packets sent to the write flow.

Without this limitation if the write flow gets backed up, there is an unlimited amount of RAM we use up in the queue of Buffers going out of the BufferPort and waiting on the input of the WriteFlow. With this limit the batching size will automatically adjust to the throughput that the WriteFlow can sustain.

### CODE REVIEW

File: src/utils/BufferPort.hxx [hunk 1]
```diff
@@ -36,6 +36,8 @@
 #ifndef _UTILS_BUFFERPORT_HXX_
 #define _UTILS_BUFFERPORT_HXX_
 
+#include "utils/LimitedPool.hxx"
+
 /// A wrapper class around a string-based Hub Port that buffersthe outgoing
```
Comment (atanisoft) [inline src/utils/BufferPort.hxx]: typo near the end of line

Comment (balazsracz) [reply src/utils/BufferPort.hxx]: fixed

File: src/utils/BufferPort.hxx [hunk 2]
```diff
@@ -171,6 +182,10 @@ private:
         BufferPort *parent_; ///< what to notify upon timeout.
     } bufferTimer_{this}; ///< timer instance.
 
+    /// Pool implementation that limits the number of buffers allocateable to
```
Comment (atanisoft) [inline src/utils/BufferPort.hxx]: allocatable

Comment (balazsracz) [reply src/utils/BufferPort.hxx]: fixed

**Comment by atanisoft** [comment]
is there an estimate on the capacity for the esp32? Should we add that to the examples?

**Comment by balazsracz** [comment]
Capacity in terms of what?

--
Sent from my mobile device

**Comment by atanisoft** [comment]
Just thinking if we should add an override to the defaults on the esp32 or not.

**Comment by balazsracz** [comment]
At the end of my work I'll have a bunch of overrides which we should push for the esp32. I'm not yet ready with them as I'm still pursuing problems.

**Pull request merged**